### DISCUSSION TITLE
[V5] Dynamic validation rules

### DISCUSSION THREAD

**Discussion by victor-sc-92** [adonisjs/core#2717] [Help] [answered]
Is there a way to use rules dynamically based on another field?
For example, when type is email, i want to use rules.email(), and when the type is cell_phone, i want to use rules.mobile().

```ts
   contacts:
      schema.array.optional([rules.minLength(1)]).members(
        schema.object().members({
          type: schema.enum(['email', 'cell_phone', 'landline'] as const),
          value: schema.string({ trim: true }),
        })
      ),
```

**Comment by arthurcvm** [comment, marked as answer]
Hi! You can use [custom rules](https://github.com/adonisjs/legacy.adonisjs.com/blob/develop/docs/docs/guides/validator/08-adding-custom-rules.md) and pass 'tip' in ValidationRuntimeOptions:
```ts
validator.rule('ruleName', async (value, _, { pointer, arrayExpressionPointer, errorReporter, tip })
```
in tip instance you have the other fields of validator, this way you can give the necessary treatment to the field.

**Reply by victor-sc-92** [reply]
Thank you, it's worked. 👍
